src/rat/core/run_sarea.py

```diff
@@ -16,13 +16,18 @@
 
 
 def run_sarea(start_date, end_date, datadir, reservoirs_shpfile, shpfile_column_dict, filt_options = None):
-    reservoirs_polygon = gpd.read_file(reservoirs_shpfile)
+    if isinstance(reservoirs_shpfile, gpd.GeoDataFrame):
+        reservoirs_polygon = reservoirs_shpfile
+    else:
+        reservoirs_polygon = gpd.read_file(reservoirs_shpfile)
     no_failed_files = 0
     Optical_files = 0
     Tmsos_files = 0
     Partial_optical_tmsos_files = 0
-    
+    i = 1
     for reservoir_no,reservoir in reservoirs_polygon.iterrows():
+        print(f"\n\n +++ PROCESSING RESERVOIR: {reservoir['tmsos_id']} - {reservoir['name_2']} ({i}/{len(reservoirs_polygon)}) +++\n\n")
+        i += 1
```

Comment:
'tmsos_id' and 'name_2' are not correct column names. It should be shpfile_column_dict['id_column'] and shpfile_column_dict['dam_name_column'].